ts/src/management/tunnelManagementHttpClient.ts

```diff
@@ -612,8 +622,22 @@ export class TunnelManagementHttpClient implements TunnelManagementClient {
         }
 
         if (!(tunnelAuthentication in headers) && tunnel?.accessTokens) {
-            for (let scope of scopes ?? []) {
-                const accessToken = tunnel.accessTokens[scope];
+            for (let scope of accessTokenScopes ?? []) {
+                let accessToken: string | null = null;
+                for (let scopeAndToken of Object.entries(tunnel.accessTokens)) {
```

Comment:
nit: can be deconstructed inline into `[scope, token]`

Reply:
I considered that, but there are so many scope and token variables already in this part of code that I thought it would be clearer to keep the key-value pair together.